src/dns_poller.c

```diff
@@ -1,5 +1,6 @@
 #include <math.h>        // NOLINT(llvmlibc-restrict-system-libc-headers)
 #include <netdb.h>       // NOLINT(llvmlibc-restrict-system-libc-headers)
+#include <string.h>      // NOLINT(llvmlibc-restrict-system-libc-headers)
```

Comment:
This file doesn't call strerror(). Why is it needed here?

Reply:
it solve some build warnings:
```
Scanning dependencies of target https_dns_proxy
[ 16%] Building C object CMakeFiles/https_dns_proxy.dir/src/logging.c.o
[ 33%] Building C object CMakeFiles/https_dns_proxy.dir/src/dns_poller.c.o
/home/teng/src/https_dns_proxy/src/dns_poller.c: In function ‘get_addr_listing’:
/home/teng/src/https_dns_proxy/src/dns_poller.c:45:14: warning: implicit declaration of function ‘strlen’ [-Wimplicit-fu
nction-declaration]
   45 |       pos += strlen(pos);
      |              ^~~~~~
/home/teng/src/https_dns_proxy/src/dns_poller.c:45:14: warning: incompatible implicit declaration of built-in function ‘
strlen’
/home/teng/src/https_dns_proxy/src/dns_poller.c:6:1: note: include ‘<string.h>’ or provide a declaration of ‘strlen’
    5 | #include "logging.h"
  +++ |+#include <string.h>
    6 |
```